feat(script): add shared utilities

byeongsu-hong opened the pull request:
* agent
    * Hyperlane agent utilities
    * Context -> HplAgentConfig. see `fromContext` for detail
    * Export function
* crypto
    * mnemonic -> privateKey
* github
    * Github Release finder
    * Release artifact downloader
* wasm
    * filepath converter
    * search created_at of specific contract (it uses rest endpoint)

byeongsu-hong committed:
feat: add shared utilities

byeongsu-hong committed:
fix: correct representation of `instant finality`

byeongsu-hong committed:
fix: also actual value

nambrot commented on script/shared/agent.ts:
```diff
@@ -0,0 +1,147 @@
+import { extractByte32AddrFromBech32 } from "./utils";
+import { getContractInfo } from "./wasm";
+
+export type HplAgentConfig = {
```
Can you import this from the SDK https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/nambrot/core-deployer-non-evm/node_modules/@hyperlane-xyz/sdk/src/metadata/agentConfig.ts#L390 ?

nambrot commented on script/shared/agent.ts:
```diff
@@ -0,0 +1,147 @@
+  network: Config["networks"][number],
+  context: Context
+): Promise<HplAgentConfig> {
+  const toHex = (v: string) => `0x${extractByte32AddrFromBech32(v)}`;
```
Sounds like this is a more general util function that should somewhere else?

nambrot commented on script/shared/agent.ts:
```diff
@@ -0,0 +1,147 @@
+  chainId: string;
+  protocol: "cosmos";
+  rpcUrls: { http: string }[];
+  grpcUrl: string;
```
Ii think this is now also an array of objects?

byeongsu-hong replied:
oh i missed that

nambrot commented on script/shared/agent.ts:
```diff
@@ -0,0 +1,147 @@
+    protocol: "cosmos",
+
+    rpcUrls: [{ http: network.endpoint.rpc }],
+    grpcUrl: network.endpoint.grpc,
```
ditto

nambrot commented on script/shared/agent.ts:
```diff
@@ -0,0 +1,147 @@
+    testRecipient: toHex(test?.msg_receiver?.address!),
+  };
+
+  return agent;
```
Consider running this through our zod type to make sure its well formatted https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/f2a0e92c7ff35ae0fef17d10ec1e2c2713578d34/typescript/sdk/src/metadata/agentConfig.ts#L211

byeongsu-hong replied:
Gotcha, I'll make this happen in following work.

nambrot commented on script/shared/github.ts:
```diff
@@ -0,0 +1,45 @@
+import { downloadFile } from "./utils";
+
+const RELEASE_API_URL =
+  "https://api.github.com/repos/many-things/cw-hyperlane/releases";
```
Can you move the repo name as a separate constant so that it is easier to override in forks?

byeongsu-hong replied:
Sure, applying.

byeongsu-hong committed:
Squashed commit of the following:

commit fad8f9e59a9de9b25028fab89c6c851d50d0dd7d
Merge: 1639e2f eb77beb
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Fri Mar 1 23:21:15 2024 +0900

    Merge branch 'eddy/pick-docs' into eddy/pick-cw-hpl

commit 1639e2fd0983c9bdc3b9814b7c8b314bd803a7d7
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Fri Mar 1 23:17:08 2024 +0900

    remove ts/sdk

commit d73647b4b073e1f142b13fd44da723cf7dcba153
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Fri Mar 1 23:15:19 2024 +0900

    Revert "Merge branch 'main' into eddy/pick-cw-hpl"

    This reverts commit 82fc10fc57ccf868ae172944c34974fa8917a48c, reversing
    changes made to 6566e7ad0ee50198c46a463369d7dfececb9865f.

commit 82fc10fc57ccf868ae172944c34974fa8917a48c
Merge: 6566e7a be64967
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Fri Mar 1 23:14:05 2024 +0900

    Merge branch 'main' into eddy/pick-cw-hpl

commit be64967876be5e8b0de21d07bd3544ef48afdf41
Author: ByeongSu Hong <hong@byeongsu.dev>
Date:   Fri Mar 1 23:08:13 2024 +0900

    doc: documentation (#95)

    * wip

    * project structure & overview

    * toc

    * swap

    * fix(schema): reflect missing contracts

    * refactor(ts): migrate sdk - 1

    * refactor(script): pull back project setup to root

    * reorg script / cli

    * fix(script): make igp deployable

    * build: redeploy

    * build: stride hyperlane deployment

    * fix(hook): default gas denom

    * feat(script): add migrate command

    * chore: detailed ignore policy

    * feat(script): supprot mnemonic

    * chore(script): cleanup

    * chore: trailing comma

    * feat: add grpc endpoint as config

    * feat: generate agent config

    * feat(script): add test-dispatch

    * chore: redundant args

    * docs: owner -> <signer>

    * build: yarn berry

    * env: yarn

    * env: add cw-hpl command

    * feat: mailbox null-check

    * fix: prune imports

    * docs: wip guide & example

    * fix: rename compose file

    * fix: use localosmosis key

    * refactor: restruct guide

    * fix: remove DATA_PATH

    * fix: add `yarn install`

    * fix: handle rest endpoint not working

    * fix: use localwasmd

    * fix: hrp

    * feat: add wallet command

    * feat: wallet generator

    * fix: use osmosis testnet

    * line spacing

    * fix: xor

    * fix: split length

    * feat: more info to replace

    * fix: osmosis testnet network name

    * chore: ignore example

    * feat: add test recipient

    * fix: trouble shooting

    * docs: done

    * docs: update README.md

    * fix: line break

    * chore: remove testnet contexts

    * wip: neutron deployment

    * feat: context -> agent config

    * feat: apply review changes (#97)

    * docs: add context / example

    * fix: real instant finality

    * fix: typo

commit e8716c39d1ef82f8b641736acb73b62c74bd2990
Author: ByeongSu Hong <hong@byeongsu.dev>
Date:   Fri Mar 1 19:57:30 2024 +0900

    review: docs (#98)

    * wip

    * project structure & overview

    * toc

    * swap

    * docs: owner -> <signer>

    * build: yarn berry

    * env: yarn

    * env: add cw-hpl command

    * docs: wip guide & example

    * fix: line break

    * fix: rename compose file

    * fix: use localosmosis key

    * refactor: restruct guide

    * fix: remove DATA_PATH

    * fix: add `yarn install`

    * fix: use localwasmd

    * fix: hrp

    * fix: use osmosis testnet

    * line spacing

    * feat: more info to replace

    * fix: osmosis testnet network name

    * chore: ignore example

    * docs: done

    * docs: update README.md

    * merge: remove conflicts

    * docs: move to root

    * docs: remove outdated docs

commit eb77bebd6b33d514be379f40af8cffedf0e1369b
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Thu Feb 29 17:55:52 2024 +0900

    docs: remove outdated docs

commit 6a600c52dde686493a1c64e6c7b6d34ba4e382c6
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Thu Feb 29 17:54:38 2024 +0900

    docs: move to root

commit 68074f013565e92d4e596f93cb2eb52eac5551db
Author: ByeongSu Hong <hong@byeongsu.dev>
Date:   Thu Feb 29 02:20:50 2024 +0900

    refactor: ts schema (#99)

    * fix(schema): reflect missing contracts

    * refactor(ts): migrate sdk - 1

commit 5896800b212821e16411e25d0c618146f62b7d4c
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Wed Feb 28 17:16:32 2024 +0900

    merge: remove conflicts

commit adb18525cf60addcb6760d6950ed4e138d410408
Author: ByeongSu Hong <hong@byeongsu.dev>
Date:   Tue Feb 27 02:59:10 2024 +0900

    docs: update README.md

commit 19c6c60101b89e5be194a12bd3f007fddbd3a6e0
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 02:57:59 2024 +0900

    docs: done

commit 5cdecaa6ae19ee1e80579774180e4b2fdd1a9f94
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 02:32:56 2024 +0900

    chore: ignore example

commit 28a5e6917cb63f7b896fc254cf35a3651e5d63d6
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 02:21:21 2024 +0900

    fix: osmosis testnet network name

commit 57e78ea2e0c6bffd27f7d971304056a7db98f232
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 02:18:23 2024 +0900

    feat: more info to replace

commit 56e2700914607485f23cf7105dd6b6aa5ba73369
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 01:35:03 2024 +0900

    line spacing

commit b707b6748bdfa7ac5c7443e90d6e84f40504b987
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 01:32:35 2024 +0900

    fix: use osmosis testnet

commit aa53d70dbf7a82cf88105aa2721b7e66713f8bf4
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 00:38:22 2024 +0900

    fix: hrp

commit bc1b84f78f4078c3cf515d455d1ebe38459680aa
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 00:36:03 2024 +0900

    fix: use localwasmd

commit 7aaa751f71c6042c25d1ecc536e85995ac39a0d0
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 00:04:08 2024 +0900

    fix: add `yarn install`

commit ca25459e023c95eb20a2f76d83ce0d510bd88b0c
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Mon Feb 26 23:56:19 2024 +0900

    fix: remove DATA_PATH

commit 82b5f59f46054381c0331b5c074a49de8f07d254
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Mon Feb 26 23:49:25 2024 +0900

    refactor: restruct guide

commit df1a67cc1a68609254f299b0883149f27f623cff
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Mon Feb 26 22:32:48 2024 +0900

    fix: use localosmosis key

commit ef171651e991faabb5b1312c12e747049cf79189
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Mon Feb 26 22:26:51 2024 +0900

    fix: rename compose file

commit 1289770ac6960a9d6db41ad12426a93565cdd9e1
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Tue Feb 27 03:00:17 2024 +0900

    fix: line break

commit 2604486e096413bd846dd11e006c490c7f4e0b08
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Mon Feb 26 22:22:49 2024 +0900

    docs: wip guide & example

commit 2b4c5a1a5072841c2cc1b0eed30e0f6a9e8e9024
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Mon Feb 26 21:21:24 2024 +0900

    env: add cw-hpl command

commit e43cfb33cb19b1a5956111d3e5bbe662b019b8f0
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Mon Feb 26 21:21:14 2024 +0900

    env: yarn

commit 76313e89fdca069d274d094d9fa4a32a6b1344a4
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Mon Feb 26 20:49:06 2024 +0900

    build: yarn berry

commit 9dba0a4d93f447092326fa0f0db6f62b702d8ec5
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Mon Feb 26 20:44:20 2024 +0900

    docs: owner -> <signer>

commit 17200dc65d2f18c43dd98afba53c9dfb4de20782
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Wed Feb 14 18:50:16 2024 +0900

    swap

commit 50b01d70caf45028cde7c80a2f46e0e0884c69e1
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Wed Feb 14 18:42:09 2024 +0900

    toc

commit 6cde4c8da4bf937cf4ad15d0138cb9ce6c1f3c2e
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Wed Feb 14 18:29:22 2024 +0900

    project structure & overview

commit cab7199fb97355bdc2f333e56d0c7e47ba635cc1
Author: byeongsu-hong <hong@byeongsu.dev>
Date:   Wed Feb 14 18:22:41 2024 +0900

    wip

byeongsu-hong committed:
merge: cw-hpl (#110)

* refactor: ts schema (#99)

* fix(schema): reflect missing contracts

* refactor(ts): migrate sdk - 1

* review: docs (#98)

* wip

* project structure & overview

* toc

* swap

* docs: owner -> <signer>

* build: yarn berry

* env: yarn

* env: add cw-hpl command

* docs: wip guide & example

* fix: line break

* fix: rename compose file

* fix: use localosmosis key

* refactor: restruct guide

* fix: remove DATA_PATH

* fix: add `yarn install`

* fix: use localwasmd

* fix: hrp

* fix: use osmosis testnet

* line spacing

* feat: more info to replace

* fix: osmosis testnet network name

* chore: ignore example

* docs: done

* docs: update README.md

* merge: remove conflicts

* docs: move to root

* docs: remove outdated docs

* doc: documentation (#95)

* wip

* project structure & overview

* toc

* swap

* fix(schema): reflect missing contracts

* refactor(ts): migrate sdk - 1

* refactor(script): pull back project setup to root

* reorg script / cli

* fix(script): make igp deployable

* build: redeploy

* build: stride hyperlane deployment

* fix(hook): default gas denom

* feat(script): add migrate command

* chore: detailed ignore policy

* feat(script): supprot mnemonic

* chore(script): cleanup

* chore: trailing comma

* feat: add grpc endpoint as config

* feat: generate agent config

* feat(script): add test-dispatch

* chore: redundant args

* docs: owner -> <signer>

* build: yarn berry

* env: yarn

* env: add cw-hpl command

* feat: mailbox null-check

* fix: prune imports

* docs: wip guide & example

* fix: rename compose file

* fix: use localosmosis key

* refactor: restruct guide

* fix: remove DATA_PATH

* fix: add `yarn install`

* fix: handle rest endpoint not working

* fix: use localwasmd

* fix: hrp

* feat: add wallet command

* feat: wallet generator

* fix: use osmosis testnet

* line spacing

* fix: xor

* fix: split length

* feat: more info to replace

* fix: osmosis testnet network name

* chore: ignore example

* feat: add test recipient

* fix: trouble shooting

* docs: done

* docs: update README.md

* fix: line break

* chore: remove testnet contexts

* wip: neutron deployment

* feat: context -> agent config

* feat: apply review changes (#97)

* docs: add context / example

* fix: real instant finality

* fix: typo

* Revert "Merge branch 'main' into eddy/pick-cw-hpl"

This reverts commit 82fc10fc57ccf868ae172944c34974fa8917a48c, reversing
changes made to 6566e7ad0ee50198c46a463369d7dfececb9865f.

* remove ts/sdk

* merge: schema refactoring (#106)

* fix(schema): reflect missing contracts

* refactor(ts): migrate sdk - 1

* feat(script): initial config setup (#102)

feat: initial config setup

* merge: docs improvements (#108)

* wip

* project structure & overview

* toc

* swap

* docs: owner -> <signer>

* build: yarn berry

* env: yarn

* env: add cw-hpl command

* docs: wip guide & example

* fix: line break

* fix: rename compose file

* fix: use localosmosis key

* refactor: restruct guide

* fix: remove DATA_PATH

* fix: add `yarn install`

* fix: use localwasmd

* fix: hrp

* fix: use osmosis testnet

* line spacing

* feat: more info to replace

* fix: osmosis testnet network name

* chore: ignore example

* docs: done

* docs: update README.md

* merge: remove conflicts

* docs: move to root

* docs: remove outdated docs

byeongsu-hong committed:
fix: apply code review
